lrauv_ignition_plugins/src/VisualizePointCloud.cc

```diff
@@ -391,82 +375,34 @@ void VisualizePointCloud::PublishMarkers()
       dataVal = this->dataPtr->floatVMsg.data(ptIdx);
     }
 
```

Comment:
I did that on purpose so that we support plain point clouds without float values. There's a comment on line 374 above, and it's also documented on the header:

https://github.com/osrf/lrauv/blob/701852c8eef41912bd162109f689bdfc30f48027/lrauv_ignition_plugins/src/VisualizePointCloud.hh#L37-L41

I know this isn't a use case we care about here, but I think it's an important use case for when we upstream the plugin, because I suspect most users won't have a separate topic publishing floats.

Reply:
Okay. I see the reason for that. I don't have a better suggestions right now.

The comments are okay for developers, but when we do upstream, we can't expect the users to read the comments, especially if this is meant to be a tool for scientists who don't care about engineering. I see that in 6f92329, you've changed to use 0.5 instead of 0. That's better for the current default colors, but, I would prefer them to be 0s actually, because 0 is a color they can control with the UI, while 0.5 is not a color they can change easily to be exactly what they want, if they don't have a float vector.

I also think the plugin UI will need to be a lot more clear about what that "Float vector" is supposed to do. Just looking at it, there's no way I'd infer that's supposed to be for per-vertex coloring. So we should put that in a TODO or an issue somewhere to make it more user-friendly before/when we upstream it.

Reply:
Oh there's a division by 0! I see. Maybe just set it to the min? 0 makes sense to me. The alternative is give them an option to specify only 1 color, if they don't have a float vector. That involves changing the UI though. In the future, it could be something like, do you want per-vertex coloring, if so, you can specify a float vector and 2 colors; otherwise, you don't get that float vector drop-down menu, and you get just 1 color. So in RViz, they expose this choice via the API - if you specify only 1 color, then they treat it as just a single color point cloud; if you specify colors[], then they treat it as per-vertex coloring.

Reply:
> The alternative is give them an option to specify only 1 color, if they don't have a float vector. That involves changing the UI though.

Ok, see 8f3b47715fa0d669bebfd994d0684332d48f1cdf, hopefully it's some improvement. When changing the UI, often it's harder to know what to change to, than to do the actual change :grimacing: 

![minmaxcolor](https://user-images.githubusercontent.com/5751272/147043697-dd67d03b-e8ca-4a0a-aebc-0208b0468dd5.gif)

> in RViz, they expose this choice via the API - if you specify only 1 color, then they treat it as just a single color point cloud

We have the same on our API since https://github.com/ignitionrobotics/ign-gui/pull/318

Reply:
This is nice, I think it's definitely smarter and gives the user some information.

Reply:
I kind of want to unresolve this so the pretty picture shows on the PR :sweat_smile:

Reply:
I copied it into the PR description :wink: